MoreLinq/DistinctBy.cs

```diff
@@ -15,16 +15,11 @@
 // limitations under the License.
 #endregion
 
-
-#if NO_HASHSET
```

Comment:
Conditional compilation on `NO_HASHSET` was there when targeting Silverlight but that's not the case so if you want to make a separate PR for removing this as well as the [alternate implementation for Silverlight](https://github.com/morelinq/MoreLINQ/blob/09aa5adcd4a10f693528f5044dd252f893fe962d/MoreLinq/DistinctBy.cs#L90-L100) then I'd be happy to merge that.